src/main/java/uk/gov/hmcts/ccd/service/MessagePublisherRunnable.java

```diff
@@ -63,10 +65,31 @@ private void processUnpublishedMessages(Pageable pageable,
         }
     }
 
+    private String getProperty(JsonNode data, String header) {
+        try {
+            return data.get(header).asText();
+        } catch (NullPointerException e) {
+            return "null";
+        }
+    }
+
     private void publishMessages(Slice<MessageQueueCandidateEntity> messagesToPublish,
                                  List<MessageQueueCandidateEntity> processedEntities) {
         messagesToPublish.get().forEach(entity -> {
-            jmsTemplate.convertAndSend(publishMessageTask.getDestination(), entity.getMessageInformation());
+            jmsTemplate.convertAndSend(
+                publishMessageTask.getDestination(),
+                entity.getMessageInformation(),
+                messagePostProcessor -> {
+                    for (String property : properties) {
+                        if (!(getProperty(entity.getMessageInformation(), property).equals("null"))) {
+                            messagePostProcessor.setStringProperty(property,
+                                getProperty(entity.getMessageInformation(), property)
+                            );
+                        }
+                    }
+                    return messagePostProcessor;
```

Comment:
is it possible to separate this into a method called `setMessageProperties` please, to make the code shorter and more readable